src/components/MonthCalendar.tsx

```diff
@@ -231,4 +141,81 @@ const MonthCalendar = ({ campaigns, events, onFocusDate, focusDate }: MonthCalen
     );
 };
 
-export default MonthCalendar;
\ No newline at end of file
+export default MonthCalendar;
+
+interface CalendarBarProps {
+    campaign: ZetkinCampaign;
+    month: number;
+    events: ZetkinEvent[];
+    firstCalendarDay: Date;
+    firstMonthDay: Date;
+    totalDaysInMonth: number;
+    gridItems: number;
+}
+
+const CalendarBar = ({ campaign, events, month, gridItems, firstCalendarDay, firstMonthDay, totalDaysInMonth }: CalendarBarProps): JSX.Element | null => {
+    const { id, color } = campaign;
+
+    const lastCalendarDay = new Date(new Date(firstCalendarDay).setDate(firstCalendarDay.getDate() + gridItems));
+
+    const barUnit = 100 / gridItems;
+
+    const getGridNumber = (event: Date) => {
+        const offset = (firstMonthDay.getDay() || 7) - 2;
+        if (event.getMonth() === month) {
+            return new Date(event).getDate() + offset;
+        }
+        if (event.getMonth() < month) {
+            return (new Date(event).getDay() || 7) - 1;
+        }
+        if (event.getMonth() > month) {
+            return (new Date(event).getDay() || 7) - 3 + offset + totalDaysInMonth;
+        }
+        return 0;
+    };
+
+    const firstEventDate = new Date(events[0].start_time);
+    const lastEventDate = new Date(events[events.length - 1].end_time);
+
+    if (firstEventDate > lastCalendarDay || lastEventDate < firstCalendarDay) {
+        return null;
+    }
+
+    let bottom, top;
+    if (firstEventDate < firstCalendarDay ) {
+        top = 0;
+    }
+    else {
+        top = getGridNumber(new Date(firstEventDate)) * barUnit;
+    }
+    if (lastEventDate > lastCalendarDay) {
+        bottom = 100;
+    }
+    else {
+        bottom = getGridNumber(new Date(lastEventDate)) * barUnit;
+    }
+    if (bottom > 100) {
+        bottom = 100;
+    }
```

Comment:
Not a big deal, and not worth changing, but I prefer to have blank lines between different sets of if/else statements. At a quick glance, it's not obvious that these are actually three separate, unrelated sets of conditionals.